### PR TITLE
Rtos/send tritium

### LINKED ISSUE #111: [RTOS] Send Tritium Task
This RTOS task is responsible for sending the drive command to the motor controller every 100 ms. The state diagrams and some pseudocode can be found on the Google Drive (Electrical Team > Controls > New Design > Software > RTOS), but don't take either resource as absolute truth. This is the time to reason about this application code, so try to think about what exactly each module should be doing.

### PR DESCRIPTION
**Pull request by Cam0Cow**
Implemented the send tritium task to work with the restructuring of the tasks.
Closes #111

### CODE REVIEW

File: Apps/Inc/Tasks.h [hunk 1]
```diff
@@ -134,4 +134,7 @@ extern OS_SEM SendCarCAN_Sem4;
  */
 // TODO: Put all global state variables here
 
+extern float RequestedMotorRPM;
+extern float RequestedMotorCurrentPercent;
+
```
Comment (JimothyGreene) [inline Apps/Inc/Tasks.h:139]: I would restructure this to go with the new CarState struct

File: Apps/Src/SendTritium.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    (void) p_arg;
+
+    OS_ERR err;
+
+    MotorController_Init();
+
+    while (1) {
+        float requested_rpm = RequestedMotorRPM;
+        float requested_current_percent = RequestedMotorCurrentPercent;
```
Comment (JimothyGreene) [inline Apps/Src/SendTritium.c:13]: Is there a need for these intermediary variables?

File: Apps/Src/Tasks.c [hunk 3]
```diff
@@ -62,3 +62,6 @@ OS_SEM SendCarCAN_Sem4;
  * Global Variables
  */
 // TODO: Put all global state variables here
+
+float RequestedMotorRPM; // Requested motor RPM (set by VelocityChange)
+float RequestedMotorCurrentPercent; // Requested motor current as fraction of max (set by VelocityChange)
```
Comment (JimothyGreene) [inline Apps/Src/Tasks.c:67]: Same as Tasks.h

**Review by JimothyGreene — CHANGES_REQUESTED** [review]

File: Apps/Inc/Tasks.h (hunk 1 above)
Comment (Cam0Cow) [reply Apps/Inc/Tasks.h:139]: The Car State struct currently holds velocity as a uint8_t, but the motor controller wants a float value. Moreover, motor controller drive command requires a value in RPM, not m/s, so I assume those fields in car state have to change

Comment (JimothyGreene) [reply Apps/Inc/Tasks.h:139]: Yeah so I would change the data types in the struct to match that. But we're just trying to get rid of these globals

File: Apps/Src/SendTritium.c [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#include "SendTritium.h"
+#include "CarState.h"
+#include <math.h>
+
+
+static float velocity_to_rpm(float velocity) {
```
Comment (JimothyGreene) [inline Apps/Src/SendTritium.c:6]: Is the naming convention for static methods snake case? No preference from me really, just make sure it's consistent.

**Review by JimothyGreene — APPROVED** [review]
Looks good to me

File: Apps/Src/SendTritium.c (hunk 4 above)
Comment (Cam0Cow) [reply Apps/Src/SendTritium.c:6]: I'm fairly sure that it's the convention that we've mostly been using. At the end of the day it doesn't really matter, the function isn't accessible outside of this file.

**Review by ErickCortez98 — APPROVED** [review]

**Pull request merged**